File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/presenters/login/PasswordRecoveryPresenter.java
package app.editors.manager.mvp.presenters.login;

import app.editors.manager.app.App;
import app.editors.manager.mvp.models.request.RequestPassword;
import app.editors.manager.mvp.models.response.ResponsePassword;
import app.editors.manager.mvp.models.response.ResponseSignIn;
import app.editors.manager.mvp.views.login.EnterpriseSSOView;
import app.editors.manager.mvp.views.login.PasswordRecoveryView;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;
import lib.toolkit.base.managers.utils.StringUtils;
import moxy.InjectViewState;

import static app.editors.manager.app.Api.PERSONAL_HOST;

@InjectViewState
public class PasswordRecoveryPresenter extends BaseLoginPresenter<PasswordRecoveryView, ResponseSignIn>{
    public static final String TAG = PasswordRecoveryPresenter.class.getSimpleName();

    protected Disposable mDisposable;

    public PasswordRecoveryPresenter() {
        App.getApp().getAppComponent().inject(this);
    }

    public void recoverPassword(String email, Boolean isPersonal) {
        if(!StringUtils.isEmailValid(email)) {
            getViewState().onEmailError();
        } else {
            sendEmailNotification(email, isPersonal);
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if(mDisposable != null) {
            mDisposable.dispose();
        }
    }

    private void sendEmailNotification(String email, Boolean isPersonal) {

        RequestPassword requestPassword = new RequestPassword();
        if(!isPersonal) {
            requestPassword.setPortal(mPreferenceTool.getPortal());
        } else {
            requestPassword.setPortal(PERSONAL_HOST);
        }
        requestPassword.setEmail(email);

        mDisposable = mRetrofitTool.getApiWithPreferences().forgotPassword(requestPassword)
                .map(ResponsePassword::getResponse)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(message -> {
                    getViewState().onPasswordRecoverySuccess(email);
                }, throwable -> {
                    getViewState().onError(throwable.getMessage());
                });

    }
}

File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/presenters/login/PasswordRecoveryPresenter.kt
package app.editors.manager.mvp.presenters.login

import app.documents.core.network.common.Result
import app.documents.core.network.common.contracts.ApiContract
import app.editors.manager.app.App
import app.editors.manager.app.accountOnline
import app.editors.manager.mvp.views.login.PasswordRecoveryView
import kotlinx.coroutines.launch
import lib.toolkit.base.managers.utils.StringUtils.isEmailValid
import moxy.InjectViewState
import moxy.presenterScope

@InjectViewState
class PasswordRecoveryPresenter : BaseLoginPresenter<PasswordRecoveryView>() {

    companion object {
        val TAG: String = PasswordRecoveryPresenter::class.java.simpleName
    }

    init {
        App.getApp().appComponent.inject(this)
    }

    fun recoverPassword(email: String, isPersonal: Boolean) {
        if (!isEmailValid(email)) {
            viewState.onEmailError()
        } else {
            sendEmailNotification(email, isPersonal)
        }
    }

    private fun sendEmailNotification(email: String, isPersonal: Boolean) {
        signInJob = presenterScope.launch {
            loginRepository.passwordRecovery(
                portal = if (isPersonal) ApiContract.PERSONAL_HOST else context.accountOnline?.portalUrl.orEmpty(),
                email = email
            ).collect { result ->
                when (result) {
                    is Result.Success -> viewState.onPasswordRecoverySuccess(email)
                    is Result.Error -> fetchError(result.exception)
                }
            }
        }
    }
}